forge/csrc/passes/erase_inverse_ops.cpp

```diff
@@ -80,6 +80,23 @@ static std::vector<graphlib::Node *> find_path_to_inverse_op(
     if (not found_inverse)
         path.clear();
 
+    // Skip bypassing the node when binary ops involved in between
+    // Ex: transpose -> multiply -> add -> transpose
+    bool is_valid_path = true;
+    int path_len = path.size();
+    if (path_len > 2)
+    {
+        for (int idx = 1; idx < path_len - 1; idx++)
+        {
+            auto &node = path.at(idx);
+            graphlib::OpNode *op_node = dynamic_cast<graphlib::OpNode *>(node);
```

Comment:
As a general note, please use `node->as<graphlib::OpNode>()` instead of manually casting to a specific node type. The `as()` method also asserts.

Additionally, if we want to change something with regards to casting in the future, we can do it at one place...